openerp/openupgrade/doc/source/development.rst

```diff
@@ -1,7 +1,34 @@
+=========================================================
 Migration script development
-++++++++++++++++++++++++++++
+=========================================================
 
 .. toctree::
    :maxdepth: 2
 
    devfaq
+
+Migration Scrips consist of two (or more) main files in every module's migration directory located in the target source code:
+
+* pre-migration.py (or pre-anythingyouwant.py)
+* post-migration.py (or post-anythingyouwant.py)
+
+What they do is derived from their names, you should check the API Documentation_ of the openupgradelib_ for details of library methods that are readily available for your migration task.
+
+The general purpose of those scripts is to manipulate directly the database entries of the source database in order to change them in a way so that they are compatible with the target database source code. To make that happen smoothly, is your task.
+When you run (TODO DOC HOWTO RUN A CUSTOM MIGRATION) a migration, Odoo's *internal* updating mechanism will be used (OpenUpgrade is providing sugar code for the migration task only). (@Experts: Under what circumstances a failing migration is reverted?)
+
+General recommendations about 'pre-migration.py' and 'post-migration.py'
+------------------------------------------------------------------------
+tbd by an expert
+
+Special Case: Module Renames and Module Merges
+----------------------------------------------
+tbd by an expert
```

Comment:
Maybe I need to be clearer about our hypothesises:
We could not put the migration code in the old module, because then we get a version conflict in the `migrating.py`. We decided to respect this version check rather than manually fool it. But it seems from the code of `openupgrade.update_module_names`, that interpreting it right, this is not the expected way to do, as this handles a query on `ir_module_module` directly, which in our case, where the module needs to be initialized manually (!Update Modules list") beforehand would result in a double entry in `ir_module_module`. I suppose we are deviating here from the standard way to do it. But as said, we get a version conflict if we copy the migration scripts simply into the old module.

Another thing, which comes to my mind, would be to simply put `openupgrade.update_module_names` in the old module migrations folder, with manually crafted versions of course and then do the actual migrating in the new module folder in the canonical way in order to trigger that that module will be pulled...

I still think a thoughtful "standard way" of doing this that is well documented, is useful.